src/test/java/org/jsmart/zerocode/core/engine/mocker/RestEndPointMockerTest.java

```diff
@@ -119,6 +135,84 @@ public void willMockASimpleGetEndPoint() throws Exception{
+                        .withStatus(mockPost.getResponse().get("status").asInt())
+                        .withHeader("Content-Type", APPLICATION_JSON)
+                        .withBody(jsonBodyResponse)));
+
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpPost request = new HttpPost("http://localhost:9073" + mockPost.getUrl());
+        request.addHeader("Content-Type", "application/json");
+        StringEntity entity = new StringEntity(bodyJson);
+        request.setEntity(entity);
+        HttpResponse response = httpClient.execute(request);
+
+        final String responseBodyActual = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        System.out.println("### response: \n" + responseBodyActual);
+
+        assertThat(response.getStatusLine().getStatusCode(), is(201));
+        JSONAssert.assertEquals(jsonBodyResponse, responseBodyActual, true);
+
     }
 
+    @Test
+    public void willMockAGetRequestWith_headers() throws Exception{
+
+        WireMock.configureFor(9073);
+
+        String jsonDocumentAsString = smartUtils.getJsonDocumentAsString("wiremock_integration/01_wire_mock_end_point_will_respond_to_get_call.json");
+        ScenarioSpec flowDeserialized = objectMapper.readValue(jsonDocumentAsString, ScenarioSpec.class);
+        MockSteps mockSteps = smartUtils.getMapper().readValue(flowDeserialized.getSteps().get(0).getRequest().toString(), MockSteps.class);
+
+        final MockStep mockGetStep = mockSteps.getMocks().get(0);
+        final Map<String, Object> headersMap = mockGetStep.getHeadersMap();
+
+        final MappingBuilder requestBuilder = get(urlEqualTo(mockGetStep.getUrl()));
+
+        // read request headers and set to request builder
+        if (headersMap.size() > 0) {
+            for (Object key : headersMap.keySet()) {
+                requestBuilder.withHeader((String) key, equalTo((String) headersMap.get(key)));
+            }
+        }
+
+        String jsonBodyResponse = mockGetStep.getResponse().get("body").toString();
+
+        stubFor(requestBuilder
+                .willReturn(aResponse()
+                        .withStatus(mockGetStep.getResponse().get("status").asInt())
+                        //.withHeader("Content-Type", APPLICATION_JSON)
+                        .withBody(jsonBodyResponse)));
+
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpGet request = new HttpGet("http://localhost:9073" + mockGetStep.getUrl());
+        request.addHeader("key", "key-007");
+        request.addHeader("secret", "secret-007");
+
+        HttpResponse response = httpClient.execute(request);
+        final String responseBodyActual = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        System.out.println("### response: \n" + responseBodyActual);
+
+        assertThat(response.getStatusLine().getStatusCode(), is(200));
+        JSONAssert.assertEquals(jsonBodyResponse, responseBodyActual, true);
+
```

Comment:
can you cover an error scenario where one of the mocks didnt mock as expected?

Reply:
done. See `org.jsmart.zerocode.wiremock.WireMockIntegrationTest`